<a href="https://colab.research.google.com/github/matiasdemori/Trabajo-Integrador-Ciencia-de-Datos/blob/main/TP_Integrados_Ciencia_de_Datos.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# TRABAJO PRÁCTICO GRUPAL INTEGRADOR
*Unidad:* 1 y 2

*Materia:* Ciencia de Datos  


---


*Fecha Entrega 1:* 27/04/2026

*Fecha Entrega 2:* 11/05/2026

### Integrantes del grupo:
* Franco Lavirgen
* Matias De Mori
* Juan Cruz Palacio Hernandez
* Angelo Grammatico

### Docentes:
* Germán Giró
* Malena Constancio
* Giselle Bustamante


# 1ra Entrega
En esta primera entrega buscamos definir el problema que vamos a analizar, comprender el contexto del proyecto y realizar una exploración inicial del dataset seleccionado siguiendo las primeras etapas de la metodología CRISP-DM. Además, planteamos nuestras preguntas analíticas principales, justificamos la elección del dataset y evaluamos de manera preliminar la estructura y calidad de los datos con los que vamos a trabajar.

##Comprensión del negocio

### Contexto
El precio de los combustibles representa una variable de gran impacto dentro de la economía argentina, ya que influye directamente en los costos de transporte, logística, producción y distribución de bienes y servicios. Además, las variaciones de precios afectan tanto a consumidores particulares como a empresas y actividades comerciales en distintas regiones del país.
Debido a esto, resulta relevante analizar cómo se comportan los precios de los combustibles según distintos factores como la ubicación geográfica, la empresa comercializadora y el tipo de producto ofrecido.

###Problema
Si bien existe información pública y actualizada sobre precios de combustibles en estaciones de servicio de Argentina, el volumen de datos y la frecuencia de actualización dificultan la identificación de patrones claros sin un análisis previo.
El problema principal consiste en comprender qué factores explican las diferencias de precios observadas en el mercado y determinar si ciertas brechas de precios se mantienen estables en el tiempo o presentan fluctuaciones significativas.

###Objetivos
<u>_Objetivo principal_</u>

- Analizar en qué medida el factor geográfico, la bandera comercial y el tipo de combustible influyen en la variabilidad de los precios promedio semanales en Argentina durante los últimos 12 meses.

<u>_Objetivos secundarios_</u>


- Comparar precios promedio semanales entre provincias.
- Analizar la dispersión de precios entre distintas empresas comercializadoras.
- Evaluar diferencias de precios entre tipos de combustible.
- Detectar regiones o empresas con precios sistemáticamente más altos o más bajos.
- Observar la estabilidad o fluctuación de las brechas de precios a lo largo del período analizado.



### Preguntas a Analizar
<u>_Pregunta analítica principal_</u>

¿En qué medida el factor geográfico, la bandera comercial y el tipo de producto explican la variabilidad de los precios semanales en Argentina, y qué tan estables se mantienen las brechas de mercado entre estos segmentos a lo largo del último año?

<u>_Preguntas analíticas secundarias_</u>

*Segmentación geográfica*

- ¿Qué provincias presentan precios promedio semanales por encima de la media nacional y qué tan amplia es la brecha entre la región más cara y la más barata?

*Competencia entre empresas*

- ¿Cómo varía la dispersión de los precios promedio semanales entre las diferentes empresas del sector, y qué proporción de marcas mantienen precios por encima de la mediana nacional de forma constante?

*Análisis por tipo de combustible*

- ¿Cuál es la magnitud de la brecha de precio entre los distintos tipos de combustible (Premium vs. Súper) y cómo ha fluctuado esta diferencia porcentual en el promedio semanal de los últimos 12 meses?
Alcance del análisis



##Comprensión de los datos

###Dataset seleccionado

- Precios en Surtidor – Resolución 314/2016
Fuente oficial: Portal de Datos Abiertos de la República Argentina.

- Contiene información sobre precios de combustibles relevados en estaciones de servicio del país.

El conjunto de datos posee una cantidad muy elevada de registros, ya que contiene información histórica desde años anteriores y actualizaciones frecuentes. Por este motivo, trabajar la base completa excede el alcance del trabajo práctico.

<u>_**Recorte definido para el análisis**_</u>

Para hacer el proyecto manejable y enfocado, se decidió trabajar con:

- Los últimos 12 meses disponibles del dataset.
- Precio promedio semanal como unidad de análisis.

Esto permite reducir el volumen de datos, disminuir ruido generado por actualizaciones horarias y enfocarse en tendencias recientes.


### Importacion de librerias necesarias
- <u>_pandas_</u>: se utilizará para leer, organizar, limpiar y analizar datos en forma de tablas, similares a una planilla de Excel.
- <u>_matplotlib.pyplot_</u>: se utilizará para generar gráficos y visualizar la información del dataset de una manera más clara.
- <u>_urllib.request_</u>: se utilizará para acceder o descargar archivos desde una URL de internet, por ejemplo un dataset en formato CSV.

In [28]:
import pandas as pd
import matplotlib.pyplot as plt
import urllib.request

- <u>_url_</u>: almacena la dirección web donde se encuentra el archivo CSV que vamos a utilizar.
- <u>_pd.read_csv(url)_</u>: lee el archivo CSV directamente desde internet.
- <u>_df_</u>: es el DataFrame donde quedan guardados los datos del archivo para poder analizarlos luego.

In [29]:
url = "http://datos.energia.gob.ar/dataset/1c181390-5045-475e-94dc-410429be4b17/resource/f8dda0d5-2a9f-4d34-b79b-4e63de3995df/download/precios-historicos.csv"

df = pd.read_csv(url)

###Describir la estructura del dataset y análisis preliminar de calidad:

Observamos las 10 primeras filas del dataset para darnos cuenta que hay caracteres del tipo "." o " " los cuales necesitan limpieza al igual que la columna *fecha_vigencia* que necesita estar apta para cambiar a *dtype:fecha*

In [30]:
df.head(10)

,idempresa,empresa,direccion,localidad,provincia,idproducto,producto,idtipohorario,tipohorario,precio,fecha_vigencia,idempresabandera,empresabandera,latitud,longitud,geojson,anio,mes
0,1376,10 DE SETIEMBRE S.A.,Av. Mosconi 299,LOMAS DEL MIRADOR,BUENOS AIRES,19,Gas Oil Grado 2,2,Diurno,985.0,01/04/2024 10:11,28,PUMA,-34.658476,-58.529443,"{""type"":""Point"",""coordinates"":[-58.529443,-34....",2024,4
1,1376,10 DE SETIEMBRE S.A.,Av. Mosconi 299,LOMAS DEL MIRADOR,BUENOS AIRES,19,Gas Oil Grado 2,3,Nocturno,985.0,01/04/2024 10:11,28,PUMA,-34.658476,-58.529443,"{""type"":""Point"",""coordinates"":[-58.529443,-34....",2024,4
2,1376,10 DE SETIEMBRE S.A.,Av. Mosconi 299,LOMAS DEL MIRADOR,BUENOS AIRES,19,Gas Oil Grado 2,2,Diurno,2190.0,01/05/2026 12:43,28,PUMA,-34.658476,-58.529443,"{""type"":""Point"",""coordinates"":[-58.529443,-34....",2026,5
3,1376,10 DE SETIEMBRE S.A.,Av. Mosconi 299,LOMAS DEL MIRADOR,BUENOS AIRES,19,Gas Oil Grado 2,3,Nocturno,2190.0,01/05/2026 12:43,28,PUMA,-34.658476,-58.529443,"{""type"":""Point"",""coordinates"":[-58.529443,-34....",2026,5
4,1376,10 DE SETIEMBRE S.A.,Av. Mosconi 299,LOMAS DEL MIRADOR,BUENOS AIRES,19,Gas Oil Grado 2,2,Diurno,1080.0,01/07/2024 09:43,28,PUMA,-34.658476,-58.529443,"{""type"":""Point"",""coordinates"":[-58.529443,-34....",2024,7
5,1376,10 DE SETIEMBRE S.A.,Av. Mosconi 299,LOMAS DEL MIRADOR,BUENOS AIRES,19,Gas Oil Grado 2,3,Nocturno,1080.0,01/07/2024 09:43,28,PUMA,-34.658476,-58.529443,"{""type"":""Point"",""coordinates"":[-58.529443,-34....",2024,7
6,1376,10 DE SETIEMBRE S.A.,Av. Mosconi 299,LOMAS DEL MIRADOR,BUENOS AIRES,19,Gas Oil Grado 2,2,Diurno,1108.0,01/08/2024 13:40,28,PUMA,-34.658476,-58.529443,"{""type"":""Point"",""coordinates"":[-58.529443,-34....",2024,8
7,1376,10 DE SETIEMBRE S.A.,Av. Mosconi 299,LOMAS DEL MIRADOR,BUENOS AIRES,19,Gas Oil Grado 2,3,Nocturno,1108.0,01/08/2024 13:40,28,PUMA,-34.658476,-58.529443,"{""type"":""Point"",""coordinates"":[-58.529443,-34....",2024,8
8,1376,10 DE SETIEMBRE S.A.,Av. Mosconi 299,LOMAS DEL MIRADOR,BUENOS AIRES,19,Gas Oil Grado 2,2,Diurno,172.5,01/10/2022 10:23,28,PUMA,-34.658476,-58.529443,"{""type"":""Point"",""coordinates"":[-58.529443,-34....",2022,10
9,1376,10 DE SETIEMBRE S.A.,Av. Mosconi 299,LOMAS DEL MIRADOR,BUENOS AIRES,19,Gas Oil Grado 2,3,Nocturno,172.5,01/10/2022 10:23,28,PUMA,-34.658476,-58.529443,"{""type"":""Point"",""coordinates"":[-58.529443,-34....",2022,10


El codigo a continuacion nos permite conocer los nombres de las columnas, la cantidad de valores no nulos en cada una y el tipo de dato de cada variable.

Esto lo utilizaremos para  describir la estructura inicial del dataset e identificar posibles problemas, como columnas con valores faltantes, variables numéricas guardadas como texto o campos que podrían necesitar una transformación antes del análisis.

In [31]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3342323 entries, 0 to 3342322
Data columns (total 18 columns):
 #   Column            Dtype  
---  ------            -----  
 0   idempresa         int64  
 1   empresa           object 
 2   direccion         object 
 3   localidad         object 
 4   provincia         object 
 5   idproducto        int64  
 6   producto          object 
 7   idtipohorario     int64  
 8   tipohorario       object 
 9   precio            float64
 10  fecha_vigencia    object 
 11  idempresabandera  int64  
 12  empresabandera    object 
 13  latitud           float64
 14  longitud          float64
 15  geojson           object 
 16  anio              int64  
 17  mes               int64  
dtypes: float64(3), int64(6), object(9)
memory usage: 459.0+ MB


Observamos que el dataset posee 18 columnas y 3342322 filas, en base a nuestras preguntas analiticas y la orientacion que le vamos a dar al proyecto solo vamos a usar las 5 columnas:

- Empresabandera
- Precio
- Producto
- Provincia
- Fecha_vigencia

En las cuales Fecha_vigencia es del
tipo object por lo tanto necesitaremos cambiar el dtype mas adelante:

Contamos la cantidad de valores nulos o vacíos que tiene cada columna del dataset.

Esto nos sirve para realizar un primer análisis de calidad de los datos, ya que nos permite identificar qué variables tienen información incompleta. A partir de este resultado podremos decidir si es necesario limpiar datos, completar valores faltantes, eliminar registros o descartar alguna variable que tenga demasiada información ausente.

In [32]:
df.isnull().sum()

,0
idempresa,0
empresa,0
direccion,0
localidad,0
provincia,0
idproducto,0
producto,0
idtipohorario,0
tipohorario,0
precio,0


Generamos un resumen estadístico de las columnas numéricas del dataset.

Esto nos sirve para analizar de manera preliminar el comportamiento de las variables cuantitativas. A partir de este resumen podemos observar valores como la media, el mínimo, el máximo y los distintos cuartiles.

También nos ayuda a detectar posibles inconsistencias o valores atípicos, por ejemplo precios demasiado altos, valores iguales a cero, registros negativos o diferencias muy grandes entre el mínimo y el máximo. Esta información es útil para definir qué variables podrían necesitar limpieza, validación o transformación antes de avanzar con el análisis.


In [33]:
df.describe()


,idempresa,idproducto,idtipohorario,precio,idempresabandera,latitud,longitud,anio,mes
count,3.342323e+06,3.342323e+06,3.342323e+06,3.342323e+06,3.342323e+06,3.340861e+06,3.340861e+06,3.342323e+06,3.342323e+06
mean,5.073572e+03,1.099478e+01,2.499988e+00,4.580390e+02,9.469797e+00,-3.352593e+01,-6.147591e+01,2.021449e+03,6.440122e+00
std,3.197128e+03,8.636993e+00,5.000001e-01,2.722729e+03,1.092387e+01,4.095867e+00,3.621483e+00,1.665761e+01,3.628330e+00
min,1.300000e+01,2.000000e+00,2.000000e+00,0.000000e+00,-1.000000e+00,-5.482950e+01,-7.233893e+01,1.000000e+00,1.000000e+00
25%,2.066000e+03,3.000000e+00,2.000000e+00,4.699000e+01,2.000000e+00,-3.479213e+01,-6.423593e+01,2.019000e+03,3.000000e+00
50%,4.852000e+03,6.000000e+00,2.000000e+00,1.374000e+02,4.000000e+00,-3.432265e+01,-6.056582e+01,2.022000e+03,6.000000e+00
75%,7.932000e+03,1.900000e+01,3.000000e+00,8.610000e+02,2.600000e+01,-3.146485e+01,-5.853706e+01,2.024000e+03,1.000000e+01
max,1.145300e+04,2.100000e+01,3.000000e+00,2.430000e+06,3.000000e+01,-2.205582e+01,-5.364826e+01,9.920000e+03,1.200000e+01


*describe* nos deja ver aspectos como\ *mean, min, max* de nuestras variables numericas, lo cual usaremos como herramienta en proximas etapas.

Calculamos tanto la cantidad como el porcentaje de valores faltantes en cada columna del dataset.

Esto nos sirve para evaluar la calidad inicial de los datos y detectar qué variables presentan información incompleta. A partir de este análisis podremos decidir, en etapas posteriores, si conviene eliminar registros, imputar valores, conservar las variables o descartarlas en caso de que tengan demasiados datos faltantes.

In [34]:
# Calculamos la cantidad y el porcentaje de valores faltantes por columna
valores_faltantes = pd.DataFrame({
    "cantidad_faltantes": df.isnull().sum(),
    "porcentaje_faltantes": df.isnull().mean() * 100
})

valores_faltantes.sort_values("porcentaje_faltantes", ascending=False)

,cantidad_faltantes,porcentaje_faltantes
geojson,1462,0.043742
longitud,1462,0.043742
latitud,1462,0.043742
idempresa,0,0.000000
empresa,0,0.000000
direccion,0,0.000000
producto,0,0.000000
localidad,0,0.000000
provincia,0,0.000000
idproducto,0,0.000000


Con este código verificamos si existen filas duplicadas dentro del dataset. Esto lo hacemos para realizar un control preliminar de calidad, ya que los registros repetidos podrían distorsionar los resultados del análisis. Si encontráramos duplicados, en una etapa posterior deberíamos evaluar si corresponde eliminarlos o conservarlos según el significado de cada registro.

In [35]:
df.duplicated().sum()

np.int64(0)

Cantidad de valores únicos por variable

Obvservamos cuántos valores distintos tiene cada columna del dataset.Para diferenciar variables categóricas, numéricas, geográficas o identificatorias. También nos ayuda a detectar posibles inconsistencias, por ejemplo categorías escritas de distintas maneras, columnas con demasiados valores únicos o variables que podrían necesitar agrupación o limpieza.


In [36]:
df.nunique().sort_values(ascending=False)

,0
fecha_vigencia,354946
precio,19794
idempresa,5698
direccion,5611
geojson,5035
latitud,5023
longitud,5022
empresa,4161
localidad,1146
anio,62


Vemos los valores únicos de las variables categóricas más importantes para nuestro análisis, como provincia, empresa y tipo de combustible.

Esto es para conocer las categorías disponibles y detectar posibles inconsistencias en la escritura de los datos. Además, nos permite validar si estas variables son adecuadas para responder nuestras preguntas analíticas relacionadas con diferencias geográficas, comerciales y por tipo de producto.

In [37]:
print(df["provincia"].unique())
print(df["empresa"].unique())
print(df["producto"].unique())

['BUENOS AIRES' 'RIO NEGRO' 'CHUBUT' 'SAN LUIS' 'SANTA FE' 'ENTRE RIOS'
 'CHACO' 'SANTIAGO DEL ESTERO' 'MENDOZA' 'CATAMARCA' 'CORDOBA' 'SALTA'
 'MISIONES' 'TUCUMAN' 'CORRIENTES' 'LA PAMPA' 'LA RIOJA' 'CAPITAL FEDERAL'
 'SANTA CRUZ' 'NEUQUEN' 'JUJUY' 'FORMOSA' 'SAN JUAN' 'TIERRA DEL FUEGO']
['10 DE SETIEMBRE S.A.' '1263 S.R.L.' '1913 SRL' ... 'ZOKFA S.A.S.'
 'ZONA CERO S.A' 'ZUSICO S.A.']
['Gas Oil Grado 2' 'Gas Oil Grado 3' 'GNC'
 'Nafta (premium) de más de 95 Ron' 'Nafta (súper) entre 92 y 95 Ron']


In [38]:
df["empresabandera"].value_counts()

,count
empresabandera,
YPF,1225658
SHELL C.A.P.S.A.,772886
AXION,522830
PUMA,318790
BLANCA,297213
DAPSA S.A.,73099
GULF,60526
REFINOR,33742
VOY,18013


Al observar los valores que toma esta columna nos damos cuenta de que blanca puede tomar dos valores:

Blanca
Sin empresa bandera
Podemos ver que hay empresas que pueden estar duplicadas ya que tuvieron un cambio de nombre, por ejemplo Puma y Petrobras.

In [39]:
df["producto"].value_counts()

,count
producto,
Nafta (súper) entre 92 y 95 Ron,802061
Gas Oil Grado 2,797300
Gas Oil Grado 3,791662
Nafta (premium) de más de 95 Ron,779144
GNC,172156


Al observar los valores de esta columna no vemos ninguna incongruencia.

In [40]:
df["provincia"].value_counts()

,count
provincia,
BUENOS AIRES,1284697
CORDOBA,407312
SANTA FE,329733
CAPITAL FEDERAL,164794
ENTRE RIOS,154852
MENDOZA,123448
RIO NEGRO,81416
MISIONES,78601
TUCUMAN,77536


**Podemos** observar viendo los valores de esta columna que se encuentran las 23 provincias y capital federal, sin duplicados.

###**Justificacion del dataset**

Se eligió este dataset porque presenta características adecuadas para responder las preguntas analíticas planteadas en el proyecto y permite trabajar distintas dimensiones relacionadas con la variabilidad de los precios de combustibles en Argentina.

Motivos principales:
* Relevancia práctica

  El precio de los combustibles representa una variable de gran impacto económico, ya que influye directamente en transporte, logística, inflación y costos operativos en distintas regiones del país.

  Analizar cómo varían los precios según la provincia, la empresa comercializadora y el tipo de combustible permite comprender mejor el comportamiento del mercado y detectar diferencias entre segmentos.

Relación con las preguntas analíticas

Las variables seleccionadas:

* *Empresa bandera*
* *Precio*
* *Producto*
* *Provincia*
* *Fecha vigencia*

Estas variables permiten abordar directamente la pregunta analítica principal del proyecto, enfocada en estudiar cómo influyen el factor geográfico, la bandera comercial y el tipo de combustible sobre la variabilidad de los precios semanales.

Además, estas variables permiten desarrollar las preguntas secundarias vinculadas a:

* Diferencias regionales entre provincias.
* Dispersión de precios entre empresas.
* Brechas de precios entre distintos tipos de combustible.

#2da Entrega

##Preparación de los datos

Realizamos un recorte temporal del dataset, tomando como referencia abril de 2026 y seleccionando los registros correspondientes a un año hacia atrás.

Este filtro fue la segmentacion planteada para el recorte de datos, haciendo que el análisis sea más manejable y esté alineado con el alcance definido para el trabajo. Además, nos permite comparar los precios de combustibles dentro de un intervalo temporal concreto, evitando mezclar datos históricos demasiado antiguos con datos actuales.

In [45]:
# Nos aseguramos de que la columna de fecha esté en formato datetime
df["fecha_vigencia"] = pd.to_datetime(df["fecha_vigencia"], errors="coerce")

# Definimos el rango de fechas que queremos analizar
fecha_inicio = "2025-04-01"
fecha_fin = "2026-04-30"

# Filtramos el dataset para quedarnos solo con los registros entre abril 2025 y abril 2026
df_recorte = df[
    (df["fecha_vigencia"] >= fecha_inicio) &
    (df["fecha_vigencia"] <= fecha_fin)
].copy()

# Verificamos el tamaño del nuevo dataset filtrado
df_recorte.shape

(351473, 18)

Con este código verificamos cuál es la fecha mínima y máxima del dataset recortado.

Esto nos sirve para confirmar que el filtro temporal se aplicó correctamente y que estamos trabajando únicamente con registros comprendidos entre abril de 2025 y abril de 2026.

In [46]:
df_recorte["fecha_vigencia"].min(), df_recorte["fecha_vigencia"].max()

(Timestamp('2025-04-01 00:00:00'), Timestamp('2026-04-30 00:00:00'))

Con este código agrupamos los registros del dataset recortado por mes.

Esto nos sirve para analizar si la cantidad de datos está distribuida de manera relativamente pareja entre abril de 2025 y abril de 2026, o si existen meses con mayor o menor cantidad de registros.

In [47]:
# Creamos una columna con el año y mes de cada registro
df_recorte["anio_mes"] = df_recorte["fecha_vigencia"].dt.to_period("M")

# Contamos la cantidad de registros por mes
registros_por_mes = df_recorte["anio_mes"].value_counts().sort_index()

# Mostramos el resultado
registros_por_mes

,count
anio_mes,
2025-04,29817
2025-05,48086
2025-06,41497
2025-07,22339
2025-08,33023
2025-09,27277
2025-10,20871
2025-11,33726
2025-12,16596


En esta etapa limpiamos el dataset quedándonos únicamente con las columnas que vamos a utilizar en nuestro análisis.

Este recorte nos sirve para simplificar el dataset, eliminar columnas que no serán utilizadas y trabajar de manera más ordenada sobre las variables directamente relacionadas con nuestras preguntas analíticas.

In [55]:
# Seleccionamos únicamente las columnas necesarias para nuestro análisis
columnas_seleccionadas = [
    "empresabandera",
    "precio",
    "producto",
    "provincia",
    "fecha_vigencia"
]

# Creamos un nuevo DataFrame solo con esas columnas
df_limpio = df_recorte[columnas_seleccionadas].copy()

# Mostramos las primeras filas del nuevo dataset
df_limpio.head()

,empresabandera,precio,producto,provincia,fecha_vigencia
10,PUMA,1733.0,Gas Oil Grado 2,BUENOS AIRES,2026-02-02
11,PUMA,1733.0,Gas Oil Grado 2,BUENOS AIRES,2026-02-02
16,PUMA,2182.0,Gas Oil Grado 2,BUENOS AIRES,2026-04-02
17,PUMA,2182.0,Gas Oil Grado 2,BUENOS AIRES,2026-04-02
22,PUMA,1237.0,Gas Oil Grado 2,BUENOS AIRES,2025-06-03


In [59]:
df_limpio["fecha_vigencia"].min(), df_limpio["fecha_vigencia"].max()

(Timestamp('2025-04-01 00:00:00'), Timestamp('2026-04-30 00:00:00'))

In [60]:
df_limpio.columns

Index(['empresabandera', 'precio', 'producto', 'provincia', 'fecha_vigencia'], dtype='object')

### Tratamiento de datos faltantes

Se realizó un análisis de valores faltantes sobre las variables seleccionadas para esta segunda etapa del trabajo práctico.

In [61]:
# Creamos una tabla con cantidad y porcentaje de valores faltantes
faltantes_df_limpio = pd.DataFrame({
    "cantidad_faltantes": df_limpio.isnull().sum(),
    "porcentaje_faltantes": df_limpio.isnull().mean() * 100
})

# Mostramos la tabla ordenada de mayor a menor porcentaje de faltantes
faltantes_df_limpio.sort_values("porcentaje_faltantes", ascending=False)

,cantidad_faltantes,porcentaje_faltantes
empresabandera,0,0.0
precio,0,0.0
producto,0,0.0
provincia,0,0.0
fecha_vigencia,0,0.0


Con este código buscamos registros donde el precio sea menor o igual a 100.

Esto nos sirve para identificar posibles inconsistencias, ya que en el contexto del análisis un precio de combustible igual a 100 o negativo no tendría sentido económico. Si aparecieran estos casos, deberían ser revisados y posiblemente excluidos o tratados en la etapa de limpieza.

In [101]:
cantidad_precios_negativos = (df_limpio["precio"] <= 100).sum()

cantidad_precios_negativos

np.int64(8)

In [102]:
# Filtramos el dataset para conservar únicamente precios mayores o iguales a 100
df_limpio = df_limpio[df_limpio["precio"] >= 100].copy()

# Verificamos el tamaño del dataset luego de la limpieza
df_limpio.shape

(351465, 6)

In [103]:
# Verificamos si todavía existen precios menores a 100
(df_limpio["precio"] < 100).sum()

np.int64(0)

Generamos un resumen de las columnas categóricas del dataset.

Esto nos sirve para conocer cuántos valores no nulos tiene cada variable, cuántas categorías distintas existen y cuál es la categoría más frecuente. Esta información es útil para comprender mejor variables como provincia, empresa o tipo de combustible, que son centrales para nuestras preguntas analíticas.

In [104]:
df_limpio.describe(include="object")

,empresabandera,producto,provincia
count,351465,351465,351465
unique,10,5,24
top,SHELL C.A.P.S.A.,Nafta (súper) entre 92 y 95 Ron,BUENOS AIRES
freq,117522,88618,148137


Luego del análisis realizado, se observó que las variables utilizadas no presentan valores faltantes.

Por este motivo, no fue necesario aplicar técnicas de imputación, eliminación de registros ni reemplazo de datos.

Esta decisión permite conservar la integridad del dataset original y trabajar únicamente sobre las transformaciones necesarias para el análisis exploratorio.

###Corrección de formatos y tipos de datos.

Con este código transformamos la columna de fecha al formato datetime de Pandas.

Esto nos sirve porque nuestro análisis se enfocará en la evolución temporal de los precios durante los últimos 12 meses. Para poder filtrar períodos, agrupar por semana y analizar tendencias, necesitamos que la fecha esté correctamente reconocida como una variable temporal.

In [105]:
df_limpio["fecha_vigencia"] = pd.to_datetime(
    df_limpio["fecha_vigencia"],
    errors="coerce"
).dt.normalize()

In [106]:
df_limpio.dtypes

,0
empresabandera,object
precio,float64
producto,object
provincia,object
fecha_vigencia,datetime64[ns]
anio_mes,period[M]


In [107]:
# Normalizamos columnas categóricas eliminando espacios al inicio y al final
columnas_texto = ["empresabandera", "producto", "provincia"]

for columna in columnas_texto:
    df_limpio[columna] = df_limpio[columna].astype(str).str.strip()

##Analisis Descriptivo

###Medidas de tendencia central y dispersión para las variables numéricas

En esta sección analizamos la variable precio, que es la principal variable numérica del dataset. Calculamos medidas de tendencia central y dispersión para conocer el comportamiento general de los precios de combustibles durante el período seleccionado.

In [108]:
df_limpio["precio"].describe()

,precio
count,351465.000000
mean,1620.784911
std,324.264859
min,121.000000
25%,1407.000000
50%,1587.000000
75%,1825.000000
max,3000.000000


Calculamos de forma separada la media, mediana, desvío estándar, valor mínimo y valor máximo del precio.

Esto nos sirve para interpretar con mayor claridad el nivel promedio de precios, su variabilidad y la diferencia entre los valores extremos del dataset.

In [109]:
# Calculamos medidas principales de la variable precio
media_precio = df_limpio["precio"].mean()
mediana_precio = df_limpio["precio"].median()
desvio_precio = df_limpio["precio"].std()
min_precio = df_limpio["precio"].min()
max_precio = df_limpio["precio"].max()

print("Media del precio:", media_precio)
print("Mediana del precio:", mediana_precio)
print("Desvío estándar del precio:", desvio_precio)
print("Precio mínimo:", min_precio)
print("Precio máximo:", max_precio)

Media del precio: 1620.78491087306
Mediana del precio: 1587.0
Desvío estándar del precio: 324.2648592508758
Precio mínimo: 121.0
Precio máximo: 3000.0


Identificamos los registros con los precios más caros y más baratos.

Esto nos sirve para observar casos extremos y analizar a qué empresas, productos y provincias pertenecen. De esta manera podemos detectar posibles diferencias relevantes entre banderas comerciales o identificar valores que podrían requerir una revisión adicional.

In [110]:
precios_mas_caros = (
    df_limpio
    .sort_values("precio", ascending=False)
    [["empresabandera", "precio", "producto", "provincia", "fecha_vigencia"]]
    .head(3)
)

precios_mas_caros



,empresabandera,precio,producto,provincia,fecha_vigencia
1185143,BLANCA,3000.0,Gas Oil Grado 3,SANTA FE,2026-04-08
1185231,BLANCA,3000.0,Nafta (premium) de más de 95 Ron,SANTA FE,2026-04-08
1185230,BLANCA,3000.0,Nafta (premium) de más de 95 Ron,SANTA FE,2026-04-08


In [111]:
precios_mas_baratos = (
    df_limpio
    .sort_values("precio", ascending=True)
    [["empresabandera", "precio", "producto", "provincia", "fecha_vigencia"]]
    .head(3)
)

precios_mas_baratos

,empresabandera,precio,producto,provincia,fecha_vigencia
1735313,SHELL C.A.P.S.A.,121.0,Nafta (súper) entre 92 y 95 Ron,BUENOS AIRES,2025-08-19
3243551,YPF,137.0,Gas Oil Grado 3,BUENOS AIRES,2025-09-21
2139472,DAPSA S.A.,141.0,Gas Oil Grado 3,BUENOS AIRES,2025-07-01


Analizamos la cantidad y el porcentaje de registros correspondientes a cada tipo de combustible.

Esto nos permite conocer qué productos tienen mayor presencia en el dataset y evaluar si la distribución de registros entre categorías es equilibrada o si algún tipo de combustible está más representado que otro.

### Tablas de frecuencia y proporciones para las variables categóricas.

In [112]:
# Calculamos frecuencia absoluta y relativa por producto
frecuencia_producto = pd.DataFrame({
    "cantidad": df_limpio["producto"].value_counts(),
    "porcentaje": df_limpio["producto"].value_counts(normalize=True) * 100
})

frecuencia_producto

,cantidad,porcentaje
producto,,
Nafta (súper) entre 92 y 95 Ron,88618,25.213890
Gas Oil Grado 2,87031,24.762352
Gas Oil Grado 3,85478,24.320487
Nafta (premium) de más de 95 Ron,82673,23.522399
GNC,7665,2.180871


Analizamos la cantidad y el porcentaje de registros por provincia.

Esto nos sirve para revisar la representación geográfica del dataset y detectar si algunas provincias concentran una mayor cantidad de observaciones que otras.

In [113]:
frecuencia_provincia = pd.DataFrame({
    "cantidad": df_limpio["provincia"].value_counts(),
    "porcentaje": df_limpio["provincia"].value_counts(normalize=True) * 100
})

frecuencia_provincia

,cantidad,porcentaje
provincia,,
BUENOS AIRES,148137,42.148436
CORDOBA,42128,11.986400
SANTA FE,30361,8.638413
CAPITAL FEDERAL,21094,6.001736
ENTRE RIOS,16931,4.817265
MENDOZA,9227,2.625297
RIO NEGRO,8287,2.357845
CORRIENTES,7892,2.245458
TUCUMAN,7642,2.174327


Analizamos la cantidad y el porcentaje de registros por empresa comercializadora.

Esto nos permite identificar qué empresas tienen mayor presencia en el dataset y evaluar si existe una concentración de registros en determinadas banderas.

In [114]:
frecuencia_empresa = pd.DataFrame({
    "cantidad": df_limpio["empresabandera"].value_counts(),
    "porcentaje": df_limpio["empresabandera"].value_counts(normalize=True) * 100
})

frecuencia_empresa

,cantidad,porcentaje
empresabandera,,
SHELL C.A.P.S.A.,117522,33.437753
YPF,70648,20.101006
AXION,68692,19.544478
PUMA,43886,12.486592
BLANCA,21972,6.251547
DAPSA S.A.,11238,3.197473
GULF,10859,3.089639
REFINOR,3285,0.934659
VOY,3207,0.912466


Buscamos posibles valores atípicos en la variable precio utilizando el método del rango intercuartílico.

Esto nos sirve para detectar precios que se encuentran muy alejados del comportamiento general de los datos. Estos casos pueden representar errores de carga, situaciones excepcionales o diferencias reales entre productos, empresas o provincias.

In [115]:
# Calculamos los cuartiles y el rango intercuartílico
Q1 = df_limpio["precio"].quantile(0.25)
Q3 = df_limpio["precio"].quantile(0.75)
IQR = Q3 - Q1

# Definimos los límites para detectar valores atípicos
limite_inferior = Q1 - 1.5 * IQR
limite_superior = Q3 + 1.5 * IQR

print("Límite inferior:", limite_inferior)
print("Límite superior:", limite_superior)

# Filtramos los registros considerados valores atípicos
outliers_precio = df_limpio[
    (df_limpio["precio"] < limite_inferior) |
    (df_limpio["precio"] > limite_superior)
]

outliers_precio

Límite inferior: 780.0
Límite superior: 2452.0


,empresabandera,precio,producto,provincia,fecha_vigencia,anio_mes
82,PUMA,2469.0,Gas Oil Grado 3,BUENOS AIRES,2026-04-02,2026-04
83,PUMA,2469.0,Gas Oil Grado 3,BUENOS AIRES,2026-04-02,2026-04
130,PUMA,599.9,GNC,BUENOS AIRES,2026-04-02,2026-04
131,PUMA,599.9,GNC,BUENOS AIRES,2026-04-02,2026-04
136,PUMA,489.9,GNC,BUENOS AIRES,2025-05-05,2025-05
...,...,...,...,...,...,...
3339763,SHELL C.A.P.S.A.,639.0,GNC,CORDOBA,2025-05-01,2025-05
3339794,SHELL C.A.P.S.A.,598.0,GNC,CORDOBA,2025-07-22,2025-07
3339795,SHELL C.A.P.S.A.,598.0,GNC,CORDOBA,2025-07-22,2025-07
3341731,PUMA,689.0,GNC,BUENOS AIRES,2025-08-06,2025-08


###Valores atípicos

Calculamos cuántos registros fueron identificados como valores atípicos y qué porcentaje representan sobre el total del dataset.

Esto nos ayuda a evaluar el impacto potencial de estos valores en el análisis. Si el porcentaje es bajo, podrían analizarse como casos particulares; si es alto, habría que revisar si responden a diferencias reales del mercado o a problemas de calidad de datos.

In [116]:
cantidad_outliers = outliers_precio.shape[0]
porcentaje_outliers = cantidad_outliers / df_limpio.shape[0] * 100

print("Cantidad de valores atípicos:", cantidad_outliers)
print("Porcentaje de valores atípicos:", porcentaje_outliers)

Cantidad de valores atípicos: 7939
Porcentaje de valores atípicos: 2.258830893545588


###Observaciones del analisis descriptivo

A partir de la exploración inicial del dataset se pudieron observar algunos patrones relevantes. En primer lugar, los registros se distribuyen de manera relativamente equilibrada entre los principales tipos de combustibles, aunque la nafta súper presenta la mayor participación dentro del conjunto de datos, mientras que el GNC posee una representación considerablemente menor.
También se identificó una fuerte concentración geográfica de registros en la provincia de Buenos Aires, seguida por Córdoba y Santa Fe, lo que indica que ciertas regiones poseen una presencia mucho más significativa dentro del dataset.
En relación con las empresas comercializadoras, Shell, YPF y Axion concentran la mayor parte de los registros analizados, lo que podría influir en los análisis comparativos entre marcas debido a la diferencia de representación respecto de otras empresas más pequeñas.
Respecto a la variable precio, se observó un promedio cercano a los 1620 pesos y un desvío estándar relativamente elevado, lo que evidencia una dispersión importante entre valores. Además, mediante el método del rango intercuartílico se detectaron aproximadamente 7.939 registros considerados atípicos, equivalentes a cerca del 2,26% del total. Estos casos podrían representar diferencias reales entre combustibles, provincias o empresas, aunque también podrían corresponder a inconsistencias o errores de carga que deberán analizarse en etapas posteriores.

##Análisis de relaciones entre variables

###Correlaciones y asociaciones entre variables.

Agrupamos los datos por tipo de combustible y calculamos el precio promedio de cada producto.

Esto nos permite observar si existen diferencias relevantes de precios entre los distintos combustibles, lo cual se vincula directamente con una de nuestras preguntas analíticas: evaluar la brecha de precios entre tipos de producto.

In [117]:
precio_por_producto = (
    df_limpio
    .groupby("producto")["precio"]
    .mean()
    .sort_values(ascending=False)
)

precio_por_producto

,precio
producto,
Gas Oil Grado 3,1763.028748
Nafta (premium) de más de 95 Ron,1741.807308
Gas Oil Grado 2,1568.140900
Nafta (súper) entre 92 y 95 Ron,1502.643985
GNC,692.809720


Agrupamos los datos por provincia y calculamos el precio promedio en cada una.

Esto nos sirve para identificar diferencias geográficas en los precios de combustibles y detectar qué provincias presentan valores promedio más altos o más bajos dentro del período analizado.

In [118]:
precio_por_provincia = (
    df_limpio
    .groupby("provincia")["precio"]
    .mean()
    .sort_values(ascending=False)
)

precio_por_provincia

,precio
provincia,
CORRIENTES,1666.912063
MISIONES,1661.537236
FORMOSA,1655.371473
SANTA FE,1653.847179
CORDOBA,1651.956835
CHACO,1651.501109
ENTRE RIOS,1650.347997
CATAMARCA,1648.092774
SANTIAGO DEL ESTERO,1644.855983


Agrupamos los datos por empresa o bandera comercial y calculamos el precio promedio de cada una.

Esto nos permite observar si existen diferencias de precios entre empresas comercializadoras y analizar si algunas banderas presentan precios sistemáticamente más altos o más bajos.

In [119]:
precio_por_empresa = (
    df_limpio
    .groupby("empresabandera")["precio"]
    .mean()
    .sort_values(ascending=False)
)

precio_por_empresa

,precio
empresabandera,
SHELL C.A.P.S.A.,1717.504027
AXION,1688.865033
VOY,1635.879576
REFINOR,1629.186423
PUMA,1623.994633
DAPSA S.A.,1615.988277
GULF,1611.029916
BLANCA,1582.927546
YPF,1405.191119


Agrupamos los datos por mes y calculamos el precio promedio mensual.

Esto nos permite observar la evolución temporal de los precios durante el período analizado y detectar si existe una tendencia creciente, decreciente o estable a lo largo del tiempo.

In [120]:
# Creamos una columna con año y mes
df_limpio["anio_mes"] = df_limpio["fecha_vigencia"].dt.to_period("M")

# Calculamos el precio promedio por mes
precio_por_mes = (
    df_limpio
    .groupby("anio_mes")["precio"]
    .mean()
)

precio_por_mes

,precio
anio_mes,
2025-04,1396.179102
2025-05,1344.172007
2025-06,1402.505946
2025-07,1460.322645
2025-08,1540.923973
2025-09,1611.722224
2025-10,1644.608364
2025-11,1733.836143
2025-12,1793.284707


Calculamos el precio promedio combinando provincia y tipo de producto.

Esto nos permite observar si las diferencias de precios entre combustibles se mantienen en todas las provincias o si existen provincias donde determinados productos presentan valores más altos o más bajos que el promedio general.

In [123]:
precio_producto_provincia = (
    df_limpio
    .groupby(["provincia", "producto"])["precio"]
    .mean()
    .reset_index()
    .sort_values("precio", ascending=False)
)

precio_producto_provincia

,provincia,producto,precio
67,NEUQUEN,Gas Oil Grado 3,1837.046317
29,CORRIENTES,Gas Oil Grado 3,1798.926000
101,SANTIAGO DEL ESTERO,Gas Oil Grado 3,1797.822599
62,MISIONES,Gas Oil Grado 3,1796.668851
12,CATAMARCA,Gas Oil Grado 3,1793.331507
...,...,...,...
23,CORDOBA,GNC,732.547532
80,SAN JUAN,GNC,656.869565
0,BUENOS AIRES,GNC,656.224034
56,MENDOZA,GNC,653.907195


Calculamos el precio promedio combinando empresa comercializadora y tipo de producto.

Esto nos permite analizar si las diferencias de precios entre empresas se mantienen para todos los combustibles o si algunas empresas presentan precios más altos en productos específicos.

In [122]:
precio_empresa_producto = (
    df_limpio
    .groupby(["empresabandera", "producto"])["precio"]
    .mean()
    .reset_index()
    .sort_values("precio", ascending=False)
)

precio_empresa_producto

,empresabandera,producto,precio
37,SHELL C.A.P.S.A.,Gas Oil Grado 3,1859.280494
38,SHELL C.A.P.S.A.,Nafta (premium) de más de 95 Ron,1849.761281
2,AXION,Gas Oil Grado 3,1841.307943
3,AXION,Nafta (premium) de más de 95 Ron,1804.848305
7,BLANCA,Gas Oil Grado 3,1795.666090
42,VOY,Gas Oil Grado 3,1788.616438
8,BLANCA,Nafta (premium) de más de 95 Ron,1772.825129
32,REFINOR,Gas Oil Grado 3,1765.412776
27,PUMA,Gas Oil Grado 3,1759.581297
28,PUMA,Nafta (premium) de más de 95 Ron,1737.539006


Para explorar si existe una relación entre el tiempo y el precio, transformamos los meses en una variable numérica y calculamos su correlación con el precio promedio mensual.

Esto nos permite tener una primera aproximación sobre si los precios tienden a aumentar o disminuir durante el período analizado.

In [124]:
# Convertimos el precio promedio mensual en un DataFrame
precio_por_mes_df = precio_por_mes.reset_index()

# Creamos una variable numérica para representar el orden de los meses
precio_por_mes_df["mes_numero"] = range(1, len(precio_por_mes_df) + 1)

# Calculamos la correlación entre el paso del tiempo y el precio promedio
correlacion_tiempo_precio = precio_por_mes_df["mes_numero"].corr(precio_por_mes_df["precio"])

correlacion_tiempo_precio

np.float64(0.9801416398269953)

A partir de estos análisis buscamos responder las preguntas planteadas en la primera entrega.

El agrupamiento por provincia permite analizar si existen diferencias geográficas en los precios y qué jurisdicciones presentan valores promedio más altos o más bajos.

El agrupamiento por empresa permite observar si hay diferencias relevantes entre banderas comercializadoras y si algunas empresas mantienen precios promedio superiores al resto.

El análisis por producto permite evaluar la brecha entre tipos de combustible, especialmente entre naftas, gasoil y GNC.

Por otra parte, el análisis mensual permite comenzar a estudiar si estas diferencias se mantienen estables en el tiempo o si presentan variaciones durante el período abril 2025 - abril 2026.

En esta etapa observamos que el precio del combustible puede analizarse en función de distintas variables categóricas, como provincia, empresa y producto. Si bien no contamos con muchas variables numéricas para aplicar correlaciones tradicionales, los promedios agrupados permiten identificar asociaciones relevantes entre el precio y los principales factores definidos en nuestras preguntas analíticas.

Estos análisis permiten avanzar sobre el objetivo central del trabajo, ya que muestran cómo varía el precio según el tipo de combustible, la ubicación geográfica, la empresa comercializadora y el período temporal.

A partir de estos resultados, en la siguiente etapa podemos complementar el análisis con visualizaciones que faciliten la comparación entre segmentos y la identificación de patrones.

###Hallazgos con las preguntas analíticas